ledger/src/blockstore_processor.rs

```diff
@@ -1850,6 +1859,8 @@ impl TransactionStatusSender {
     ) {
         let slot = bank.slot();
 
+        let hash = self.bank_hash.then(|| bank.hash_internal_state());
```

Comment:
I like the idea of getting mid-slot information; however, I have a few concerns with this approach.

This function takes a `Vec<SanitizedTransaction>`; there is logic elsewhere in this file that determines how those are broken up. Two point about that:
- We'll have details (ie the bank hash) after several transactions that get lumped together, not a single tx
- While there are some rules about what can't be in one batch (see below snippet), there is a variable definition of what is acceptable. For example, suppose a slot has 3 non-conflicting transactions. Executing these in one batch OR in three separate batches are both valid from the protocol's perspective, but these will cause conflicting output in this file. The batching logic could change release-to-release without breaking consensus, or, this file could potentially get used to compare to other clients
https://github.com/solana-labs/solana/blob/d00c9a45b2c2b6d9f0c137bb5a572e1afde2b4aa/ledger/src/blockstore_processor.rs#L586-L633

Rather, I think a different approach could be making ledger tool support a mode where it executes tx-by-tx; this is something I've done before for debugging (recently), and something that Alessandro expressed interest to me in having.

Also, I think taking the bank hash here is overkill. While it does encapsulate the state, it is expensive to compute, and I think it is abusing the concept of the bank hash a little. Generally, we talk about bank hashes for a block that has been fully replayed (ie all transactions) as opposed to partially.

Lastly, I think including this in the `TransactionStatusSender` isn't the right place. As the touched files in the diff would show, this type propagates to a number of places that are in the "production" validator codepath (`BankingStage`, `RPC`, etc). So, while it is convenient to tuck into here, the hash details is a debug feature and I'd like to keep it as self-contained as possible

Reply:
This makes a lot of sense. For this PR, I haven't added the transactions yet.